projects/packages/sync/src/modules/class-posts.php

```diff
@@ -434,6 +444,12 @@ public function filter_post_content_and_add_links( $post_object ) {
 			$post->post_password = 'auto-' . wp_generate_password( 10, false );
 		}
 
+		// Explicitly truncate post_content when it exceeds limit.
+		// Large content will cause OOM issues and break Sync.
+		if ( strlen( $post->post_content ) >= self::MAX_POST_CONTENT_LENGTH ) {
+			$post->post_content = '';
```

Comment:
This may be a stupid question, but wouldn't we want to sync part of the content at least?

Reply:
Actually this is the question I'm toying with. We could always just get the first x bytes of reasonable length, or the first x paragraphs. My concern is that we split it so the content is no longer valid (uneven tags).

I'm leaning on the full truncate as it gives no ambiguity to why the cache site and remote db are not the same. Oh post_content is empty -> it was over the limit.

Reply:
> My concern is that we split it so the content is no longer valid (uneven tags).

Ah yes, that's a good point. I would assume we could also have opening shortcodes (such as shortcodes used by page builders) and no closing shortcode. 

Considering this, not syncing anything probably makes more sense. 👍